root/static/scripts/genre/components/GenreEditForm.js

```diff
@@ -26,17 +40,88 @@ type Props = {
   +form: GenreFormT,
 };
 
+/* eslint-disable flowtype/sort-keys */
+type ActionT =
+  | {+type: 'update-name', +action: NameActionT};
+/* eslint-enable flowtype/sort-keys */
+
+type StateT = {
+  +form: GenreFormT,
+  +guessCaseOptions: GuessCaseOptionsStateT,
+  +isGuessCaseOptionsOpen: boolean,
+  +previousForm?: GenreFormT | null,
+};
+
+type WritableStateT = {
+  ...StateT,
+  form: WritableGenreFormT,
+  guessCaseOptions: WritableGuessCaseOptionsStateT,
+  previousForm?: GenreFormT | null,
+};
+
+function createInitialState(form: GenreFormT) {
+  return {
+    form,
+    guessCaseOptions: createGuessCaseOptionsState(),
+    isGuessCaseOptionsOpen: false,
+  };
+}
+
+function reducer(state: StateT, action: ActionT): StateT {
+  return mutate<WritableStateT, StateT>(state, newState => {
```

Comment:
For simple reducers like this you can forgo using `mutate` if you want, since it only really helps for deeply-nested assignments, like say `newState.form.field.name = 'foo';`.  (IIRC, I actually didn't use it at all in the new relationship editor code.)
```suggestion
  const newState = {...state};
  // ...
  return newState;
```

Reply:
This actually complains in `runNameReducer` so since it works I'd leave it as-is (it also seems more consistent for other editors which might need `mutate` more)